src/components/CustomDropdown/CustomDropdown.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect, useRef } from "react";
+import PropTypes from "prop-types";
+import "./CustomDropdown.css";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCaretDown } from "@fortawesome/free-solid-svg-icons";
+import { FormError } from "../FormError/FormError";
+import SearchIcon from "../../assets/search-icon.svg";
+
+export const CustomDropdown = (props) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedOption, setSelectedOption] = useState(props.defaultValue);
+  const [searchOption, setSearchOption] = useState("");
+  const searchInputRef = useRef(null);
+
+  const clampedMinValue = Math.max(0, props.minValue);
+  const clampedMaxValue = Math.min(99, props.maxValue);
+
+  const handleOptionClick = (option) => {
+    setSelectedOption(option);
+    props.onChange(option);
+    setIsOpen(false);
+  };
+
+  const handleSearch = (event) => {
+    setSearchOption(event.target.value);
+  };
+
+  useEffect(() => {
+    if (isOpen) {
+      searchInputRef.current.focus();
+    }
+  }, [isOpen]);
+
+  const numberOptions = [];
+  for (let i = clampedMinValue; i <= clampedMaxValue; i++) {
+    numberOptions.push({ id: i.toString(), value: i.toString() });
+  }
+
+  const combinedOptions = [...numberOptions, ...(props.options || [])];
+
+  return (
+    <>
+      <div className={`dropdown ${isOpen ? "open" : ""}`}>
+        <button
+          className={`${isOpen ? "ds-rounded-t-[4px]" : "ds-rounded-[4px]"} ${
+            props.hasError ? "ds-border-[#D3080C]" : "ds-border-[#6f6f66]"
+          } dropdown-select ds-border-2`}
+          onClick={() => setIsOpen(!isOpen)}
+        >
+          {isOpen ? (
+            <div className={`ds-flex ${props.hasSearch ? "" : "ds-hidden"}`}>
+              <img src={SearchIcon} alt="search icon" />
+              <input
+                type="text"
+                value={searchOption}
+                ref={searchInputRef}
+                onChange={handleSearch}
+                placeholder="Search..."
+                className="input-search"
+              />
+            </div>
+          ) : (
+            selectedOption
+          )}
+          <FontAwesomeIcon
+            icon={faCaretDown}
+            style={{ color: "#666666" }}
+            className="ds-absolute ds-right-0 ds-px-[14px]"
+          />
+        </button>
+        <div className="dropdown-open">
+          {isOpen && (
+            <ul className="dropdown-options">
+              {combinedOptions
```

Comment:
In future iterations, this should be a seperate function.

Reply:
This is a good suggestion, thanks. Updated the code to separate the string and number options.